rfd/0127-resource-versioning.md

```diff
@@ -0,0 +1,34 @@
+---
+authors: Edoardo Spadolini (edoardo.spadolini@goteleport.com)
+state: draft
+---
+
+# RFD 0127 - Resource versioning
+
+## Required Approvers
+
+- Engineering: TBD ("@r0mant || @zmb3 || @russjones"?)
+
+## What
+
+To define best practices around updating resources and choosing versions for them.
+
+## Why
+
+So far, all resources have used a single increasing version number (generally `v2` or `v3` with legacy `v1` resources not defined in protobuf form but referring to some older bespoke format that's no longer in use or supported - but with exceptions like `network_restrictions` having jumped from `v1` directly to `v4`, or `role` supporting up to `v6`). That, combined with the custom of renaming the message type used to store it, leads to frequent renaming of struct types across the entire codebase for no benefit other than to be able to know at a glance that the `FooV6` struct is intended to hold `foo` resources of version up to `v6` (but with no idea of how low of a version is actually supported). This RFD proposes a more useful versioning scheme with fewer tradeoffs.
+
+## Details
```

Comment:
Would there be any changes in the user UX? One thing we should avoid is making users think about which resource version to pick and e.g. remember to specify `version: v6.15` or something. Would we default to latest minor if someone just creates a resource `version: v6` for example?

Can you add a separate UX section that explicitly covers details like this please.